Firestore/core/test/unit/remote/bloom_filter_test.cc

```diff
@@ -132,6 +143,153 @@ TEST(BloomFilterTest,
   }
 }
 
+class BloomFilterGoldenTest : public ::testing::Test {
+ public:
+  void RunGoldenTest(const std::string& test_file) {
+    BloomFilter bloom_filter = GetBloomFilter(test_file);
+    std::string membership_result = GetMembershipResult(test_file);
+
+    for (size_t i = 0; i < membership_result.length(); i++) {
+      bool expectedResult = membership_result[i] == '1';
+      bool mightContainResult =
+          bloom_filter.MightContain(kGoldenDocumentPrefix + std::to_string(i));
+
+      EXPECT_EQ(mightContainResult, expectedResult);
+    }
+  }
+
+ private:
+  static const char* kGoldenDocumentPrefix;
+  JsonReader reader;
+
+  static Path GetGoldenTestFolder() {
+    return Path::FromUtf8(__FILE__).Dirname().AppendUtf8(
+        "bloom_filter_golden_test_data/");
+  }
+
+  json ReadFile(const std::string& file_name) {
+    Path file_path = GetGoldenTestFolder().AppendUtf8(file_name);
+    std::ifstream stream(file_path.native_value());
+    HARD_ASSERT(stream.good());
+    return nlohmann::json::parse(stream);
+  }
+
+  BloomFilter GetBloomFilter(const std::string& file_name) {
```

Comment:
nit: Rename `GetBloomFilter()` to `ReadBloomFilter()` or `LoadBloomFilter()` or something that more clearly implies that it is loading the bloom filter from an external resource.